Alluvial/DataStreamCatchup{T}.cs

```diff
@@ -77,26 +79,39 @@ internal class DataStreamCatchup<TData> : IDataStreamCatchup<TData>
                                 .ContinueWith(async t =>
-                                                    {
-                                                        var batch = t.Result;
-
-                                                        if (batch.Count > 0)
-                                                        {
-                                                            var aggregatorUpdates =
-                                                                aggregatorSubscriptions
-                                                                    .Values
-                                                                    .Select(subscription => Aggregate(stream.Id,
-                                                                                                      (dynamic) subscription,
-                                                                                                      batch,
-                                                                                                      streamQuery.Cursor))
-                                                                    .Cast<Task>();
-
-                                                            await Task.WhenAll(aggregatorUpdates);
-                                                        }
-                                                    });
+                                {
+                                    var batch = t.Result;
+
+                                    if (batch.Count > 0)
+                                    {
+                                        var aggregatorUpdates =
+                                            aggregatorSubscriptions
+                                                .Values
+                                                .Select(subscription => Aggregate(stream.Id,
+                                                                                  (dynamic) subscription,
+                                                                                  batch,
+                                                                                  streamQuery.Cursor))
+                                                .Cast<Task>();
+
+                                        try
+                                        {
+                                            await Task.WhenAll(aggregatorUpdates);
+                                        }
+                                        catch (Exception exception)
+                                        {
+                                            aggregatorException = exception;
+                                            throw;
+                                        }
+                                    }
+                                });
                         });
 
                 await Task.WhenAll(batches);
+
+                if (aggregatorException != null)
+                {
+                    throw aggregatorException;
+                }

```

Comment:
Exceptions from aggregator updates are unobserved unless I marshal the exception to the outer scope like this. This seems strange to me and this change feels like a workaround.